network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractIdentifiableImpl.java

```diff
@@ -327,18 +321,16 @@ public <E extends Extension<I>> boolean removeExtension(Class<E> type) {
         if (extension == null) {
             return false;
         }
-        AtomicBoolean removed = new AtomicBoolean(false);
         index.notifyExtensionBeforeRemoval(extension);
-        updateResource(r -> removed.set(r.getAttributes().getExtensionAttributes().remove(extension.getName()) != null));
-        if (removed.get()) {
```

Comment:
why remove this if ?

Reply:
In the previous implementation of removeExtension(): 
1. getExtension() => check that the extension is in the map
2. if not, we return false
3. remove the extension from the map 
4. if the extension existed in the map, return true, otherwise return false.  

I think the `removed` boolean was duplicative, since we already check whether the extension exists in the map.

In the PR implementation:
1. getExtension() => check that the extension is in the map
2. if not, we return false
3. remove the extension from the map 
4. return true

I do the remove and return true at the end, assuming that if the extension exists, we'll be able to remove it. 

Alternatively, my other idea was to make index.removeExtensionAttributes() return a boolean? Even though I think that in the end it will always return true if you've first checked that the extension is in the map?